middleware/auth_bearer.go

```diff
@@ -36,30 +37,105 @@ func BearerAuthFuncGRPC(auth authentication.Authentication) grpc_auth.AuthFunc {
+type ClaimInjectorJWT func(ctx context.Context, token jwt.Claims) (context.Context, error)
+
+// ClaimInjectorBehavior is used in combination with ClaimInjectorJWT when grouping
+// different claim injectors by using GroupClaimInjectors.
+type ClaimInjectorBehavior func(ctx context.Context, err error) (context.Context, error)
+
+// groupClaimInjectors treats all the given injectors as a single one.
+//
+// By setting a mandatoryInjection, the resulting injector will early return
+// if an error is found at any point during the claim injection.
+//
+// If optionalInjection is used instead, no errors will be returned during the
+// claim injection.
+//
+// If no ClaimInjectorBehavior is provided, it defaults to mandatoryInjection.
+//
+// Example:
+//
+//	groupClaimInjectors(mandatoryInjection,
+//		groupClaimInjectors(mandatoryInjection, SubjectClaimer),
+//		groupClaimInjectors(optionalInjection, EmailClaimer),
+//	)
+//
+// For public-accessible methods, check GroupMandatoryClaimInjectors and
+// GroupOptionalClaimInjectors.
+func groupClaimInjectors(behavior ClaimInjectorBehavior, injectors ...ClaimInjectorJWT) ClaimInjectorJWT {
+	if behavior == nil {
+		behavior = mandatoryInjection
+	}
+	return func(ctx context.Context, token jwt.Claims) (context.Context, error) {
+		for _, injector := range injectors {
+			var err error
+			ctx, err = behavior(injector(ctx, token))
+			if err != nil {
+				return ctx, err
+			}
+		}
+		return ctx, nil
+	}
+}
 
+// mandatoryInjection forces a ClaimInjectorJWT to always return an error.
+func mandatoryInjection(ctx context.Context, err error) (context.Context, error) {
+	if err != nil {
+		return ctx, err
 	}
+	return ctx, nil
+}
+
+// optionalInjection ignores any errors returned by a ClaimInjectorJWT.
+func optionalInjection(ctx context.Context, _ error) (context.Context, error) {
+	return ctx, nil
+}
+
+// GroupMandatoryClaimInjectors allows injecting mandatory ClaimInjectors as a single one.
+// Check groupClaimInjectors to understand how grouping works.
+func GroupMandatoryClaimInjectors(injectors ...ClaimInjectorJWT) ClaimInjectorJWT {
+	return groupClaimInjectors(mandatoryInjection, injectors...)
+}
+
+// GroupOptionalClaimInjectors allows injecting optional ClaimInjectors as a single one.
+// Check groupClaimInjectors to understand how grouping works.
+func GroupOptionalClaimInjectors(injectors ...ClaimInjectorJWT) ClaimInjectorJWT {
+	return groupClaimInjectors(optionalInjection, injectors...)
 }
 
-// injectClaims injects claims from token into the given context.
-// It returns the context with the claims injected, or an error if the claims could not be obtained.
-// Values injected:
-//   - Subject (mandatory): jwt.Claims.
-//   - Email address (optional): authentication.EmailClaimer.
-func injectClaims(ctx context.Context, token jwt.Claims) (context.Context, error) {
+// SubjectClaimer is a ClaimInjectorJWT for the "sub" claim.
```

Comment:
This documentation isn't particularly useful if you don't fully understand what an Injector is. It would help to add a more practical explanation.

```suggestion
// SubjectClaimer is a ClaimInjectorJWT that extracts the "sub" claim from an incoming JWT token and stores it in the request context.
```